task_2/join.go

```diff
@@ -1,71 +0,0 @@
-package task_2
-
-import (
-	"context"
-	"fmt"
-	"math/rand"
-	"sync"
-	"time"
-)
-
-func Start() {
-
-	var (
-		numOfChannels = rand.Intn(12) + 1
-		channels      []chan interface{}
-	)
-
-	fmt.Println(fmt.Sprintf("creating %d channels..", numOfChannels))
-	for i := 0; i < numOfChannels; i++ {
-		channels = append(channels, make(chan interface{}))
-	}
-
-	ctx, _ := context.WithTimeout(context.Background(), 5*time.Second)
-
-	joinedChannel := join(ctx, channels...)
-	for i, c := range channels {
-		c <- fmt.Sprint("hello_", i)
-	}
-
-	for {
-		select {
-		case <-time.After(10 * time.Second):
-			fmt.Println("reached time limit waiting for joined output, exiting.")
-			return
-		case joinedOutput, ok := <-joinedChannel:
-			if !ok {
-				return
-			}
-			fmt.Println("joined output: ", joinedOutput)
-		}
-	}
-
-}
-
-func join(ctx context.Context, channels ...chan interface{}) chan interface{} {
-	joinedChannel := make(chan interface{})
-	var wg sync.WaitGroup
```

Comment:
Wait group is ok. You can also use approach described in the book you read (Concurrency in go) by using recursion or use dynamic select (please google it :D) though reflect package.